src/main/java/io/github/kawamuray/wasmtime/Trap.java

```diff
@@ -4,7 +4,7 @@
 import lombok.NonNull;
 
 @AllArgsConstructor
-public class Trap {
+public class Trap extends RuntimeException {
```

Comment:
I think `Trap` class should remain as a direct mapping of `wasmtime::Trap` in Java side. 
Shall we introduce an exception class for trap occurrence and let it hold a `Trap` instance as a field like this?


```java
public class TrapException extends WasmtimeException {
    private final Trap trap;
...
}
```